TrackletAlgorithm/MatchEngine.h

```diff
@@ -301,6 +301,9 @@ void MatchEngine(const BXType bx, BXType& bx_o,
 			  istub = 0;
 			  tail_readindex++;
 			}
+			else if (istep == kMaxProc - kMaxProcOffset(module::ME) - 1) { // The last loop iteration
```

Comment:
is this logically equivalent to the clearing of the multi-D arrays in the VMRouter?

Reply:
yes-ish... but here it keeps track of from which memory address the ME should READ from. Since it only reads from one memory at a time, it's not a multi-dimensional array.

The reason I clear it in the last loop-iteration (in contrast to the VMR where I do it in the first) is just pure trial and error... I don't understand what HLS seems to understand and not. To me it doesn't make much sense.